### PR TITLE
Email to balloter instead of logged in user

### LINKED ISSUE #169: [Student Portal | Lockers] Send an email to user of ballot when ballot has been cancelled
Catches accidental or malicious deletion of locker ballot by other users

### LINKED ISSUE #170: [Student Portal | Lockers] Send an email to user of ballot when ballot has been edited
To catch accidental or malicious editing of ballot by other users.

Currently the current logged in user is emailed. This does not notify the user of the ballot in the event that the ballot edited does not belong to the current logged in user.

### PR DESCRIPTION
**Pull request by unusep**
Fixes #169, fixes #170 

(Also fixes indentation)

### CODE REVIEW

File: app/controllers/portal/locker/locker_ballots_controller.rb [hunk 1]
```diff
@@ -1,75 +1,76 @@
...
 
-	def destroy
-		if is_authenticated_user
-      @ballot.destroy
-      redirect_to portal_locker_locker_ballots_path, notice: 'Your Ballot has been cancelled'
-    else
-      show_error_message_at portal_locker_locker_ballots_path
+    def destroy
+        if is_authenticated_user
+            mail_notify(@ballot.user, @ballot)
```
Comment (leeyimin) [inline app/controllers/portal/locker/locker_ballots_controller.rb]: More changes to mailers/ballot_notifier.rb and its views are probably required to fix #169. Current changes will send an email telling users a ballot was submitted when it has been deleted.

Comment (unusep) [reply app/controllers/portal/locker/locker_ballots_controller.rb]: Oh right. My bad. Will update it.

Comment (unusep) [reply app/controllers/portal/locker/locker_ballots_controller.rb]: Okay updated it.

File: app/mailers/ballot_notifier.rb [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+        @ballot = ballot
+        mail(to: Rails.application.secrets.locker_mail_to, reply_to: 'connect@nuscomputing.com', subject: '[Locker Ballot] Ballot Received') if user.email
+    end
+
+    def deleted_ballot_to_user(submitter, user, ballot)
+        @round = ballot.round
+        @submitter = submitter
+        @user = user
+        @ballot = ballot
+        mail(to: user.email, reply_to: 'connect@nuscomputing.com', subject: '[Locker Ballot] Ballot Rescinded') if user.email
```
Comment (leeyimin) [inline app/mailers/ballot_notifier.rb:31]: The mail job is not done before the ballot is deleted from db, resulting in a failed print job. Hmm I think we just dont print information about the ballot? @jia1

Comment (unusep) [reply app/mailers/ballot_notifier.rb:31]: I fixed it by passing to mail the lockerlocation and ballotround information directly. Will have to eventually add test cases for regression testing.

**Pull request merged**